client/components/domains/register-domain-step/index.jsx

```diff
@@ -560,6 +561,21 @@ const RegisterDomainStep = React.createClass( {
 					severity = 'info';
 				}
 				break;
+			case 'tld_in_maintenance':
+				tldIndex = domain.lastIndexOf( '.' );
+				if ( tldIndex !== -1 ) {
+					message = this.translate(
+						'Domains ending with {{strong}}%(tld)s{{/strong}} are undergoing maintenance. They will be available again soon.',
```

Comment:
I think it is important, as users today do understand what maintenance is, while the `temporary unavailable` is treated like an error, or at least that's my perception. :) I like your 2nd part and my 1st part. :)